Vagrantfile

```diff
@@ -26,28 +26,36 @@ end
 
 Vagrant.configure(2) do |config|
 
-    config.vm.box = "debian/jessie64"
+    # Prefer docker over virtualbox since it is listed first
+    # meaning that docker should run if no '--provider' is supplied
+    # when calling vagrant up
+    config.vm.provider "docker" do |d, configOverride|
+        d.build_dir = "."
+        d.pull = true
+        d.has_ssh = true
+
+        # Overrides for 'config' unique for docker
+        configOverride.ssh.username = "vagrant"
+        configOverride.ssh.password = "vagrant"
+    end
 
-    config.vm.provider "virtualbox" do |vb|
+    config.vm.provider "virtualbox" do |vb, configOverride|
```

Comment:
@jonte I've started to try and find out internally about how prepared we are to support this, let's see what comes up. In the meantime, we really would benefit from having a Docker provider so for me it's fine to merge this as it is and perhaps change later.